nucypher/network/server.py

```diff
@@ -248,10 +248,10 @@ def reencrypt_via_rest(self, id_as_hex, request: Request):
             capsule_signed_by_both = bytes(self._stamp(capsule_signature))
 
             capsule.set_correctness_keys(verifying=alices_verifying_key)
-            # TODO: Sign the result of this.  See #141.
             cfrag = pre.reencrypt(kfrag, capsule, metadata=capsule_signed_by_both)
             self.log.info("Re-encrypting for {}, made {}.".format(capsule, cfrag))
-            cfrag_byte_stream += VariableLengthBytestring(cfrag)
+            signature = self._stamp(bytes(cfrag))
```

Comment:
I think we should tie the `CFrag` with the `Capsule` in this signature somehow. Otherwise, you only know that the `CFrag` corresponds to some `Capsule` in the `WorkOrder` but you don't know which. I'd say something in the spirit of ` signature = self._stamp(bytes(capsule) + bytes(cfrag))`.